dtfs-central-api/src/types/db-models/tfm-user.ts

```diff
@@ -0,0 +1,11 @@
+import { WithId } from 'mongodb';
+import { Team } from '../tfm/teams';
+
+export type TfmUser = WithId<{
+  username: string;
+  email: string;
+  teams: Team[];
+  timezone: string;
+  firstName: string;
+  lastName: string;
+}>;
```

Comment:
As discussed, the type of this user for incoming requests from TFM is slightly different to what we have in the DB. The DB has these fields:
<img width="196" alt="image" src="https://github.com/UK-Export-Finance/dtfs2/assets/16437079/4bda5bb7-289e-43e9-aa43-fdbd3ed4e35a">
but the incoming user from TFM has a subset of these, based on the below function in the TFM API
```
const mapUserData = (user) => ({
  username: user.username,
  firstName: user.firstName,
  lastName: user.lastName,
  email: user.email,
  timezone: user.timezone,
  teams: user.teams,
  lastLogin: user.lastLogin,
  status: user.status,
  _id: user._id,
});
```
so we should have a type outside of db-models to represent this.

Note in this type I think `lastLogin` is of type `number` and `_id` is of type `string`.

Reply:
Discussed: we are going to keep the `tfm-user.d.ts` `db-model` type, but define a new `tfm-session-user.d.ts` type which omits certain properties from the DB model type